core/tests/protractor_desktop/profileFeatures.js

```diff
@@ -164,3 +168,61 @@ describe('Visiting user profile page', function() {
     general.checkForConsoleErrors([]);
   });
 });
+
+describe('playing the exploration', function() {
+  var TEST_USERNAME = 'testUser';
+  var TEST_EMAIL = TEST_USERNAME + '@example.com';
+
+  var continueButton = element(by.css('.protractor-test-continue-button'));
+  var backButton = element(by.css('.protractor-test-back-button'));
+  var nextButton = element(by.css('.protractor-test-next-button'));
+
+  var explorationPlayerPage = null;
+  var libraryPage = null;
+
+  var EXPLORATION = {
+    title: 'A new exploration',
+    category: 'Learning',
+    objective: 'The goal is to create a new exploration',
+    language: 'English'
+  };
+
+  beforeAll(function() {
+    users.createUser(TEST_EMAIL, TEST_USERNAME);
+    users.login(TEST_EMAIL);
+    workflow.createAndPublishTwoCardExploration(
+      EXPLORATION.title,
+      EXPLORATION.category,
+      EXPLORATION.objective,
+      EXPLORATION.language
+    );
+    libraryPage = new LibraryPage.LibraryPage();
+    explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
+    libraryPage.get();
+    libraryPage.findExploration(EXPLORATION.title);
+    libraryPage.playExploration(EXPLORATION.title);
+  });
```

Comment:
This block of code will be executed for all the ```it``` tests written inside this describe block. Do we really want all of this happening for all the tests written inside this block? Maybe it will be better to move your test specific code inside the ```it``` block.

Reply:
@Showtim3 There is only one it test inside the describe block.

Reply:
Yes but I was considering the need if in future any other test needs to be added in this describe block. Also take a look at the comment above. You probably forgot to add the ```afterEach``` block in the describe you created.

Reply:
@Showtim3  This describe block says `playing the exploration`. So I think we have to play an exploration before every *it* test?

Reply:
@nishantwrp If there are more `it` blocks, then it will probably cause issues down the line.